### DISCUSSION TITLE
Why search cost show linear growth when more client try to call search for milvus cluster?

### DISCUSSION THREAD

**Discussion by xiaojuan8** [milvus-io/milvus#16022] [Q&A and General discussion]
Hi, I am testing milvus cluster performance. I found that when I increase number of my clients. Cost of search request is growing linearly. But I have 5 query nodes in my cluster as shown below:
![image](https://user-images.githubusercontent.com/74544828/157834646-e2415cba-f902-4bd3-84b7-9a4b2dcf8229.png)

my params:
----------------------------------
build index:
index type: IVFPQ
index embedding num: 112w 
dim : 32 (float)
nlist : 2 * sqrt(112w)
m: 8
----------------------------------
search:
nprobe: 16
topk: 1000
NQ: 10000
----------------------------------
when searching: segment size = 3
----------------------------------
node server info:
cores : 32
node container mem: 62 Gi
cache_size: 50 Gi
---------------------------------

when I start more clients, their cost performance shows below(I start the same client with same code):
![image](https://user-images.githubusercontent.com/74544828/157835797-fc4b37bd-b869-469b-b01c-2168b6d4b3eb.png)
![image](https://user-images.githubusercontent.com/74544828/157835836-cc65ffa0-8938-4836-857b-32611e2b7046.png)
![image](https://user-images.githubusercontent.com/74544828/157835865-53c1f06f-a731-4a99-8fcd-add2967fe39e.png)

**Comment by xiaofan-luan** [comment]
Because the embedding number is fairly small and only 1-2 segments is used, which means only 1-2 querynode is involved in the test.

On the otherside your nq is large, which request more computing resources. 

So it has nothing to do with how many clients you had, but your server side cpu is full.

To solve the problem you would need memory replicas. For smaller dataset we can build multiple replicas and the query load will be spread to multiple querynode.

By the way, what is the qps you are looking for? If you want to learn more about performance tuning you can email me at xiaofan.luan@zilliz.com

**Reply by xiaojuan8** [reply]
Hi, thanks for your reply. So, how to config "memory replicas" for milvus cluster with helm? Is there any reference docs ? 
BTW,  I just want to achieve the performance in milvus benchmark : https://milvus.io/cn/docs/v2.0.x/benchmark.md
